libcxx/include/__type_traits/result_of.h

```diff
@@ -25,7 +25,9 @@ template <class _Callable>
 class _LIBCPP_DEPRECATED_IN_CXX17 result_of;
 
 template <class _Fp, class... _Args>
-class _LIBCPP_TEMPLATE_VIS result_of<_Fp(_Args...)> : public __invoke_of<_Fp, _Args...> {};
+class _LIBCPP_TEMPLATE_VIS result_of<_Fp(_Args...)> {
+  using type = __invoke_result_t<_Fp, _Args...>;
```

Comment:
Perhaps we shouldn't simplify `result_of` because the member `type` was required to be conditionally present in C++14/17, see [N3462](https://wg21.link/n3462).